contrib/RL_out_sharp.lua

```diff
@@ -0,0 +1,208 @@
+    * configure other export options (size, etc.)
+    * export, images will be first exported in the temp format, then sharpened
+    * sharpened images will be stored in jpg format in the output folder
+
+  EXAMPLE
+    set sigma = 0.7, iterations = 10, jpeg output quality = 95,
+    that will be sufficient for most uses
+
+  CAVEATS
+    MAC compatibility not tested
+  
+  BUGS, COMMENTS, SUGGESTIONS
+    send to Marco Carrarini, marco.carrarini@gmail.com
+
+  CHANGES
+    * 20200308 - initial version
+  
+]]
+
+local dt = require "darktable"
+local du = require "lib/dtutils"
+local df = require "lib/dtutils.file"
+local dtsys = require "lib/dtutils.system"
+
+-- module name
+local MODULE_NAME = "RL_out_sharp"
+
+-- check API version
+du.check_min_api_version("5.0.2", MODULE_NAME)  -- darktable 3.x
+
+-- OS compatibility
+local PS = dt.configuration.running_os == "windows" and  "\\"  or  "/"
+
+-- translation
+local gettext = dt.gettext
+gettext.bindtextdomain(MODULE_NAME, dt.configuration.config_dir..PS.."lua"..PS.."locale"..PS)
+local function _(msgid)
+  return gettext.dgettext(MODULE_NAME, msgid)
+  end
+
+-- initialize module preferences
+if not dt.preferences.read(MODULE_NAME, "initialized", "bool") then
+  dt.preferences.write(MODULE_NAME, "sigma", "string", "0.7")
+  dt.preferences.write(MODULE_NAME, "iterations", "string", "10")
+  dt.preferences.write(MODULE_NAME, "jpg_quality", "string", "95")
+  dt.preferences.write(MODULE_NAME, "initialized", "bool", true)
+  end 
+
+
+-- setup export ---------------------------------------------------------------
+local function setup_export(storage, img_format, image_table, high_quality, extra_data)
+  -- set 16bpp if format is tif
+  if img_format.extension == "tif" then
+    img_format.bpp = 16
+    end
+  end
+
+
+-- temp export formats: jpg and tif are supported -----------------------------
+local function supported(storage, img_format)
+  return (img_format.extension == "jpg") or (img_format.extension == "tif")
+  end
+
+
+-- export and sharpen images --------------------------------------------------
+local function export2RL(storage, image_table, extra_data) 
+
+  local temp_name, new_name, run_cmd, result
+  local input_file, output_file, options
+
+  -- read parameters
+  local gmic = df.sanitize_filename(dt.preferences.read(MODULE_NAME, "gmic_exe", "string"))
+  local output_folder = output_folder_selector.value
+  local sigma_str = string.gsub(string.format("%.2f", sigma_slider.value), ",", ".")
+  local iterations_str = string.format("%.0f", iterations_slider.value)
+  local jpg_quality_str = string.format("%.0f", jpg_quality_slider.value)
+
+  -- save preferences
+  dt.preferences.write(MODULE_NAME, "sigma", "string", sigma_str)
+  dt.preferences.write(MODULE_NAME, "iterations", "string", iterations_str)
+  dt.preferences.write(MODULE_NAME, "jpg_quality", "string", jpg_quality_str)
+
+  local gmic_operation = " -deblur_richardsonlucy "..sigma_str..","..iterations_str..",1"
```

Comment:
I used fx_unsharp_richardsonlucy and built my widgets to resemble the gui in Gmic_qt in GIMP.  This adds the choice of exponential blur (0) or guassian blur (1) and a cut parameter that seems to preserve luminance (If I understand correctly it restricts the output betweeen 0 and 255 for each iteration).  fx_unsharp_richardsonlucy calls deblur_richardsonlucy and maps the arguments.

Reply:
I can use that, no pb, but the GMic functions of Gimp plugin are undocumented and I read somewher that they might change without notice. I thought using -deblur_richardsonlucy was safer.
When coming to preserve luminance, you might have a point. In fact, I have to force cut to 0,255 at the end of the pipeline (see GMic command line). Can't say which is better.
Is -fx_unsharp_richardsonlucy compatible with tiff input files ? I will have to test

Reply:
I wrote a postsharpen script that doesn't restrict the file types used.  I didn't try RL against all the file types so I'm not sure if there is a problem with them or not.  I guess I need to check.

It's [here](https://github.com/wpferguson/extra-dt-lua-scripts/blob/master/postsharpen.lua)

fx_unsharp_richardsonlucy just calls deblur_richardsonlucy to do all the work.

After thinking about it I would say that using deblur_richardsonlucy is probably a better choice since it is part of the api and not something built on top of it.